pkg/serializer/serializer.go

```diff
@@ -72,7 +72,7 @@ func initExtraHeaders() {
 
 // MetricSerializer represents the interface of method needed by the aggregator to serialize its data
 type MetricSerializer interface {
-	SendEvents(e marshaler.Marshaler) error
+	SendEvents(e marshaler.StreamJSONMarshalerFactory) error
```

Comment:
This interface does not feel generic at all 😢. What do you think about using `Events` instead ?
Pro: It avoid creating an interface that is confusing if you don't have full context
Con: It's not aligned with the rest, makes testing harder

Reply:
`MetricSerializer` is quite a partial abstraction as it take interfaces but you have to call the right methods. In my case, the interface is here only to make unit tests easier and so I will rename it to  `EventsMarshaler`.